src/gwt/src/org/rstudio/studio/client/workbench/BrowseAddinsDialog.java

```diff
@@ -149,20 +151,27 @@ public void onError(ServerError error)
             Debug.logError(error);
          }
          
-         private Command onSuccess_;
+         private CommandWithArg<RAddins> onSuccess_;
          
       };
      
       // first call for cached addins then call for full reindex
-      server_.getRAddins(false, new AddinsServerRequestCallback(new Command() {
+      server_.getRAddins(false, new AddinsServerRequestCallback(new CommandWithArg<RAddins>() {
          @Override
-         public void execute()
+         public void execute(RAddins addins)
          {
-            server_.getRAddins(true, new AddinsServerRequestCallback());
+            addins_ = addins;
```

Comment:
Couldn't this assignment be taken care of within the shared onResponseReceived handler? (thus eliminating the need to use CommandWithArg)